include/intx/int128.hpp

```diff
@@ -63,46 +63,56 @@ struct uint<128>
 {
     static constexpr unsigned num_bits = 128;
 
-    uint64_t lo = 0;
-    uint64_t hi = 0;
+    uint64_t words[2]{};
 
     constexpr uint() noexcept = default;
 
-    constexpr uint(uint64_t high, uint64_t low) noexcept : lo{low}, hi{high} {}
+    constexpr uint(uint64_t hi, uint64_t lo) noexcept : words{lo, hi} {}
 
     template <typename T,
         typename = typename std::enable_if_t<std::is_convertible<T, uint64_t>::value>>
-    constexpr uint(T x) noexcept : lo(static_cast<uint64_t>(x))  // NOLINT
+    constexpr uint(T x) noexcept : words{static_cast<uint64_t>(x), 0}  // NOLINT
     {}
 
 #ifdef __SIZEOF_INT128__
-    #pragma GCC diagnostic push
-    #pragma GCC diagnostic ignored "-Wpedantic"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
     constexpr uint(unsigned __int128 x) noexcept  // NOLINT
-      : lo{uint64_t(x)}, hi{uint64_t(x >> 64)}
+      : words{uint64_t(x), uint64_t(x >> 64)}
     {}
 
     constexpr explicit operator unsigned __int128() const noexcept
     {
         using builtin_uint128 = unsigned __int128;
-        return (builtin_uint128{hi} << 64) | lo;
+        return (builtin_uint128{words[1]} << 64) | words[0];
```

Comment:
Btw, why not use `lo_half/hi_half` in all these helpers?

Reply:
Helpers are only used in templates.